extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.js

```diff
@@ -0,0 +1,437 @@
+					return acc.replace('__' + key + '__', args[key])
+				}, message);
+			}
+
+			this.getText = function (notification) {
+				if (notification.type === notificationTypes.discussionReply) {
+					return this._getReplyText(notification);
+				} else if (notification.type === notificationTypes.discussionUpvotePost) {
+					return this._getPostUpvoteText(notification)
+				} else if (notification.type === notificationTypes.discussionUpvoteReply) {
+					return this._getReplyUpvoteText(notification);
+				} else {
+					return notification.title;
+				}
+			};
+
+			this._getReplyText = function (notification) {
+				const key = this._getReplyKey(notification.title, notification.totalUniqueActors);
+				const message = window.mw.message(key).parse();
+				var args = {
+					postTitle: bold(notification.title)
+				};
+
+				if (notification.totalUniqueActors > 2) {
+					args.mostRecentUser = notification.latestActors[0].name;
+					args.number = notification.totalUniqueActors - 1;
+				} else if (notification.totalUniqueActors == 2) {
+					args.firstUser = notification.latestActors[0].name;
+					args.secondUser = notification.latestActors[1].name;
+				} else {
+					args.user = notification.latestActors[0].name;
+				}
+
+				return fillArgs(message, args);
+			};
+
+			this._getReplyKey = function (title, totalUniqueActors) {
+				const user = totalUniqueActors <= 1 ? '' :
+					totalUniqueActors == 2 ? 'two-users-' : 'multiple-users-';
+				return 'notifications-replied-by-' + user + (title ? 'with-title' : 'no-title');
+			};
+
+			this._getPostUpvoteText = function (notification) {
+				const key = 'notifications-post-upvote' + this._getUpvoteKey(notification.title, notification.totalUniqueActors);
+				const message = window.mw.message(key).parse();
+				return fillArgs(message, {postTitle: bold(notification.title)});
+			};
+
+			this._getReplyUpvoteText = function (notification) {
+				const key = 'notifications-reply-upvote' + this._getUpvoteKey(notification.title, notification.totalUniqueActors);
+				const message = window.mw.message(key).parse();
+				return fillArgs(message, {postTitle: bold(notification.title)});
+			};
+
+			this._getUpvoteKey = function (title, totalUniqueActors) {
+				return '-' + (totalUniqueActors <= 1 ? 'single-user' : 'multiple-users')
+					+ '-' + (title ? 'with-title' : 'no-title');
+			};
+		}
+
+		function View(logic, template, textFormatter) {
+			this.logic = logic;
+			this.template = template;
+			this.textFormatter = textFormatter;
+			this.$notificationsCount = $('#on-site-notifications-count');
+			this.$container = $('#notification-container');
+			this.$markAllAsReadButton = $('#mark-all-as-read-button');
+
+			this.registerEvents = function () {
+				this.addDropdownLoadingEvent();
+				this.addMarkAllAsReadEvent();
+			};
+
+			this.addMarkAllAsReadEvent = function () {
+				this.$markAllAsReadButton.click($.proxy(this.logic.markAllAsRead, this.logic));
```

Comment:
`proxy()` is passé. In modern world we use `this.logic.markAllAsRead.bind(this.logic)`